src/Tools/IdeBenchmarks/InheritanceMargin/InheritanceMarginGlyphBenchmarks.cs

```diff
@@ -0,0 +1,220 @@
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using BenchmarkDotNet.Attributes;
+using Microsoft.CodeAnalysis.Editor.Host;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.LanguageServices.Implementation.InheritanceMargin;
+using Microsoft.VisualStudio.LanguageServices.Implementation.InheritanceMargin.MarginGlyph;
+using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+using Moq;
+
+namespace IdeBenchmarks.InheritanceMargin
+{
+    [MemoryDiagnoser]
+    public class InheritanceMarginGlyphBenchmarks
+    {
+        private const int MemberCount = 500;
+        private const int Iterations = 10;
+        private const double WidthAndHeightOfGlyph = 18;
+
+        private readonly UseExportProviderAttribute _useExportProviderAttribute = new();
+        private readonly IWpfTextView _mockTextView;
+
+        private Application _wpfApp;
+        private Canvas _canvas;
+        private ImmutableArray<InheritanceMarginTag> _tags;
+        private IThreadingContext _threadingContext;
+        private IStreamingFindUsagesPresenter _streamingFindUsagesPresenter;
+        private ClassificationTypeMap _classificationTypeMap;
+        private IClassificationFormatMap _classificationFormatMap;
+        private IUIThreadOperationExecutor _operationExecutor;
+        private IAsynchronousOperationListener _listener;
+
+        public InheritanceMarginGlyphBenchmarks()
+        {
+            _wpfApp = null!;
+            _threadingContext = null!;
+            _streamingFindUsagesPresenter = null!;
+            _classificationTypeMap = null!;
+            _classificationFormatMap = null!;
+            _operationExecutor = null!;
+            _listener = null!;
+            _canvas = null!;
+            var mockTextView = new Mock<IWpfTextView>();
+            mockTextView.Setup(textView => textView.ZoomLevel).Returns(100);
+            _mockTextView = mockTextView.Object;
+        }
+
+        [GlobalSetup]
+        public Task SetupAsync() => SetupWpfApplicaitonAsync();
+
+        [IterationSetup]
+        public Task IterationSetupAsync()
+        {
+            _useExportProviderAttribute.Before(null);
+            return PrepareGlyphRequiredDataAsync(CancellationToken.None);
+        }
+
+        [IterationCleanup]
+        public void IterationCleanup()
+        {
+            _useExportProviderAttribute.After(null);
+        }
+
+        [GlobalCleanup]
+        public void Cleanup()
+        {
+            RunOnUIThread(() => _wpfApp.Shutdown());
+        }
+
+        [Benchmark]
```

Comment:
💡 A baseline benchmark method should also be added which performs all the operations except for creating the `InheritanceMarginGlyph` itself

Reply:
I am guessing you mean the other benchmark I added? BenchmarkInheritanceMarginServiceAsync
That one measures the InheritanceMarginItem creation part